### PR TITLE
handlebars: Fix superfluous line breaks in ConcatStatement

### PR DESCRIPTION
**Pull request by dcyriller**
Fix https://github.com/prettier/prettier/issues/6252 and improve (fix?) https://github.com/prettier/prettier/issues/6207

- [x] I’ve added tests to confirm my change works.
- [x] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: tests/handlebars-concat-statement/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -122,3 +110,54 @@ singleQuote: true
...
+=====================================output=====================================
+<div
+  class="a very long list of classes that exceeds {{eighty
+    }} chars with emtpy spaces"
+>
+  hey
+</div>
+<a
+  href="a-very-long-href-from-a-third-party-marketing-platform{{id
+    }}longer-than-eighty-chars"
```
Comment (evilebottnawi) [inline tests/handlebars-concat-statement/__snapshots__/jsfmt.spec.js.snap]: I think we should avoid moving on next line in this case, it is ugly

Comment (dcyriller) [reply tests/handlebars-concat-statement/__snapshots__/jsfmt.spec.js.snap]: 👍 I will update, thanks for the feedback

Comment (dcyriller) [reply tests/handlebars-concat-statement/__snapshots__/jsfmt.spec.js.snap]: Will be fixed once https://github.com/prettier/prettier/pull/7052 is rebased

Comment (dcyriller) [reply tests/handlebars-concat-statement/__snapshots__/jsfmt.spec.js.snap]: No wait, it won't. I will have to work a bit more on this issue.

Comment (dcyriller) [reply tests/handlebars-concat-statement/__snapshots__/jsfmt.spec.js.snap]: It should be good now. But we need to settle #7052 before.

File: changelog_unreleased/handlebars/pr-7051.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  Hello
+</div>
+
+{{!-- Prettier master --}}
+<div
+  class="hello {{if goodbye true}} {{if goodbye false}} {{if goodbye true}} {{if
+      goodbye
+      false
+    }} {{if goodbye true}}"
+>
```
Comment (evilebottnawi) [inline changelog_unreleased/handlebars/pr-7051.md]: Maybe better :
```hbs
<div
  class="hello {{if goodbye true}} {{if goodbye false}} {{if goodbye true}} {{
    if goodbye false
  }} {{if goodbye true}}"
>
```

/cc @prettier/core what do you think about that?

Comment (dcyriller) [reply changelog_unreleased/handlebars/pr-7051.md]: Also, should:
```
<p>
  Hi {{firstName}} {{lastName}} , welcome!
</p>
```
be formatted to (with a short max-width):
```
<p>
  Hi {{firstName}} {{lastName
  }} , welcome!
</p>
```
or to:
```
<p>
  Hi {{firstName}} {{
    lastName
  }} , welcome!
</p>
```

Comment (evilebottnawi) [reply changelog_unreleased/handlebars/pr-7051.md]: Second example better

**Comment by velrest** [comment]
Hey, thanks for your work! On the playground for this pr i still get this weird formatting:

__Original__
```hbs
<button
  class="uk-padding-remove uk-button uk-button-default uk-button-small uk-width-1-{{intl.locales.length}} {{if (contains locale intl.locale) "uk-button-primary"}}"
>
  test
</button>
```
__Prettier Output__
```hbs
<button
  class="uk-padding-remove uk-button uk-button-default uk-button-small uk-width-1-{{intl.locales.length}} {{if
      (contains locale intl.locale)
      "uk-button-primary"}}"
>
  test
</button>
```
Is this intended?

**Comment by dcyriller** [comment]
Weird indeed but at least not broken! 😉

I’ve not had time recently to improve that by implementing @evilebottnawi  comment, but it is on top of my list.

File: changelog_unreleased/handlebars/pr-7051.md (hunk 2 above)
Comment (dcyriller) [reply changelog_unreleased/handlebars/pr-7051.md]: Here we go: https://github.com/prettier/prettier/pull/7157 🙂

Comment (dcyriller) [reply changelog_unreleased/handlebars/pr-7051.md]: Here we go: #7157 🙂

**Comment by dcyriller** [comment]
CI will be 🍏 after https://github.com/prettier/prettier/pull/7072 https://github.com/prettier/prettier/pull/7156 and https://github.com/prettier/prettier/pull/7157 are settled.

@velrest I've found the time, you can preview the changes in the last commit 🙂 (playground shouldn't be okay for now).

**Comment by Alonski** [comment]
@evilebottnawi Can this please be merged and released :)

**Comment by dcyriller** [comment]
Thank you @lipis and @evilebottnawi 🙏

**Comment by evilebottnawi** [comment]
Need fix CI problems

**Comment by dcyriller** [comment]
It seems to be a random CI failure (on NodeJs installation).

**Comment by dcyriller** [comment]
friendly ping @evilebottnawi I think this might be ready

**Comment by velrest** [comment]
Sorry for the long wait.

I opened https://github.com/ember-template-lint/ember-template-lint/pull/1074 for the `--help` and cli options parsing.

**Pull request merged**